### PR TITLE
DOMTokenList: separate whitespace trim/duplicate removal; add versions

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
In `DOMTokenList`, we have an entry about how modifications will trim the whitespace and remove duplicates, with partial implementation notes for IE and Edge about how they trim whitespace but not remove duplicates.  However, from my testing, I've found that Chrome and Firefox also implemented whitespace trimming before duplicate removal.  This PR separates the subfeature into two subfeatures, and adds/fixes versions for both of them.

### CODE REVIEW

**Comment by foolip** [comment]
This will remove an entry, so adding the needs-release-note label.

**Comment by foolip** [comment]
@vinyldarkscratch can you link to the tests you used for this?

**Comment by vinyldarkscratch** [comment]
Certainly!  I used the following pieces of JavaScript to test manually in BrowserStack:


`trim_whitespace`:
```js
var elm = document.createElement('b'); elm.className = ' foo bar foo '; elm.classList.remove('bar'); return elm.className === 'foo foo' || elm.className === 'foo';
```

`remove_duplicates`:
```js
var elm = document.createElement('b'); elm.className = ' foo bar foo '; elm.classList.remove('bar'); return elm.className === 'foo';
```

**Comment by foolip** [comment]
@vinyldarkscratch the `remove_duplicates` seems to assume that if duplicates are removes then whitespace is also trimmed. That's probably true, but did you check what the `elm.className` was in the latest build where it *wasn't* 'foo'? A value like ' foo  foo ' is not impossible to imagine in some implementation where these things were done in the other order.

**Comment by vinyldarkscratch** [comment]
I checked all the browser versions, and have confirmed that white space trimming came first or at the same time in all the browsers, and that duplicate removal had come later.

File: api/DOMTokenList.json [hunk 1]
```diff
@@ -527,21 +527,19 @@
           }
         }
       },
-      "remove_whitespace_and_duplicates": {
+      "remove_duplicates": {
         "__compat": {
-          "description": "Trimming of whitespace and removal of duplicates",
+          "description": "Removal of duplicates",
```
Comment (ddbeck) [inline api/DOMTokenList.json:532]:
```suggestion
          "description": "Removes duplicates",
```

File: api/DOMTokenList.json [hunk 2]
```diff
@@ -820,6 +816,55 @@
           }
         }
       },
+      "trim_whitespace": {
+        "__compat": {
+          "description": "Trimming of whitespace",
```
Comment (ddbeck) [inline api/DOMTokenList.json:821]:
```suggestion
          "description": "Trims whitespace",
```

**Review by ddbeck — CHANGES_REQUESTED** [review]
I'll leave the review of the data to Philip, but I had a couple nitpicky style suggestions. Thank you!

**Review by foolip — APPROVED** [review]

**Pull request merged**